File: 프로그래머스/unrated/181866. 문자열 잘라서 정렬하기/문자열 잘라서 정렬하기.java
//1. 일반 풀이
import java.util.*;
// class Solution {
//     public String[] solution(String myString) {
//         String[] s = myString.split("x"); Arrays.sort(s);
//         List<String> answer = new LinkedList<>(); for(String str : s) if(!str.equals("")) answer.add(str); return answer.stream().toArray(String[]::new);
//     }
// }

//2. 스트림 풀이
class Solution {
    public String[] solution(String myString){
        return Arrays.stream(myString.split("x")).sorted().filter(s->!s.equals("")).toArray(String[]::new);
    }
}

File: 프로그래머스/0/181866. 문자열 잘라서 정렬하기/문자열 잘라서 정렬하기.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

vector<string> solution(string myString) {
    vector<string> answer;
    string word = "";
    for(int i = 0; i < myString.size(); i++){
        if(myString[i] == 'x'){
            if(!word.empty()){
            answer.push_back(word);
            word = "";
            }
        }else{
            word += myString[i];
        }
    }
    if(word != ""){
        answer.push_back(word);
    }
    sort(answer.begin(), answer.end());
    return answer;
}